Google.Api.Generator/Utils/Typ.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Google.Api.Generator.Utils
+{
+    /// <summary>
+    /// Abstraction of a .NET type.
+    /// </summary>
+    internal abstract class Typ
```

Comment:
The reason I chose `Typ` is because this is used extensively throughout the code and I wanted it to be very concise.
I agree it may not be immediately clear what it is, but the conciseness was more important.

`ClrType` I suppose may be more suitable, but it's over twice the number of characters!!

Reply:
Hmm. I definitely found it distracting to see Typ (and typ in comments) all over the place. It always made me think it was a mistake.
I guess you're going to be the one reading and writing this code the most, but it still makes me sad. I'd *at least* add a comment to explain the name oddity.

Reply:
Done. Comment and TODO added.